test/testapp.go

```diff
@@ -106,6 +111,30 @@ func (a *testApp) lastAppliedTheme() fyne.Theme {
 	return a.appliedTheme
 }
 
+func (a *testApp) transitionCloud(p fyne.CloudProvider) {
+	err := p.Setup(a)
+	if err != nil {
+		fyne.LogError("Failed to set up cloud provider "+p.ProviderName(), err)
+		return
+	}
+	a.cloud = p
+
+	listeners := a.prefs.ChangeListeners()
+	if pp, ok := p.(fyne.CloudProviderPreferences); ok {
+		a.prefs = pp.CloudPreferences(a)
+	} else {
+		a.prefs = internal.NewInMemoryPreferences()
+	}
+
+	for _, l := range listeners {
+		a.prefs.AddChangeListener(l)
+		l() // assume that preferences have changed because we replaced the provider
+	}
+
+	// after transition ensure settings listener is fired
+	a.settings.apply() //just a test - "cedric"
```

Comment:
I am not sure I need to be in the comment for posterity here :-)

Reply:
oops!